presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java

```diff
@@ -91,12 +90,12 @@ public LocationHandle forTemporaryTable(SemiTransactionalHiveMetastore metastore
                 DIRECT_TO_TARGET_NEW_DIRECTORY);
     }
 
-    private LocationHandle createLocationHandle(HdfsContext context, ConnectorSession session, Path targetPath, TableType tableType, boolean tempPathRequired)
+    private LocationHandle createLocationHandle(HdfsContext context, ConnectorSession session, Path targetPath, TableType tableType, boolean tempPathRequired, Optional<Path> externalLocation)
     {
         Optional<Path> tempPath = tempPathRequired ? Optional.of(createTemporaryPath(session, context, hdfsEnvironment, targetPath)) : Optional.empty();
-        if (shouldUseTemporaryDirectory(session, context, targetPath)) {
+
+        if (shouldUseTemporaryDirectory(session, context, targetPath, externalLocation)) {
             Path writePath = createTemporaryPath(session, context, hdfsEnvironment, targetPath);
-            createDirectory(context, hdfsEnvironment, writePath);
```

Comment:
Is the directory created in createTemporaryPath()?

Reply:
Yes, I corrected what I thought was a mistake. 
The `writePath` directory is created at the end of `createTemporaryPath`

Trino made the correction before.
Am I wrong ?

Reply:
You are right.

Reply:
LGTM. Thanks